src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java

```diff
@@ -481,7 +482,11 @@ private ByteBuffer current() throws IOException {
                     if (debug.on()) debug.log("Next Buffer");
                     currentBuffer = currentListItr.next();
                 } catch (InterruptedException ex) {
-                    // continue
+                    try {
+                        close();
+                    } catch (IOException ignored) {
+                    }
```

Comment:
This will swallow the interrupt status, you' need to do Thread.currentThread().interrupt() before throwing the IOException.